src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -456,6 +460,29 @@ get_object_id(MonoObject *obj)
 	return ref->id;
 }
 
+static void
+assembly_loaded (MonoProfiler *prof, MonoAssembly *assembly)
+{
+	DEBUG_PRINTF (2, "assembly_loaded callback called for %s\n", assembly->aname.name);
+	MonoImage *assembly_image = assembly->image;
+	MonoImage *pdb_image = NULL;
+	MonoDebugHandle *handle = mono_debug_get_handle (assembly_image);
+	if (handle) {
+		MonoPPDBFile *ppdb = handle->ppdb;
+		pdb_image = mono_ppdb_get_image (ppdb);
+	}
+
+	if (pdb_image) {
```

Comment:
if the assembly has and embedded pdb are we sending it over the wire twice?

Reply:
We probably also want to either pass the value of mono_has_pdb_checksum as an arg to mono_wasm_asm_loaded or skip calling it entirely if it is false.

Reply:
> if the assembly has and embedded pdb are we sending it over the wire twice?

good point. @thaystg can you take these?

Reply:
I'm working on it, pushing a commit soon.

Reply:
Yes, we are sending twice, inside the dll and the data outside the dll, we don't need to send the pdb outside if it's embedded.

Reply:
@lewing Now we check if it's an embedded PPDB and in this case send only the assembly, and we check if it's a assembly without debug information and in this case we don't send anything.